src/python-common/ceph/deployment/service_spec.py

```diff
@@ -2009,6 +2030,13 @@ def validate(self) -> None:
                     f'Cannot add ingress: Invalid health_check_interval specified. '
                     f'Valid units are: {valid_units}')
 
+        # validate SSL parametes
+        if self.monitor_ssl:
+            if not self.ssl:
+                raise SpecValidationError('To enable SSL for stats, SSL must also be enabled on the frontend.')
+            if any([self.monitor_ssl_cert, self.monitor_ssl_key]) and not all([self.monitor_ssl_cert, self.monitor_ssl_key]):
```

Comment:
I think this line is equivalent to:

`if bool(self.monitor_ssl_cert) != bool(self.monitor_ssl_key):` 

As it checks if any of them is enabled (and not the other)